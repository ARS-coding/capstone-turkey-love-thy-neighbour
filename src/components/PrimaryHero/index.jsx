import React from "react";

import { useTranslation } from "react-i18next";

import { Container, Row, Col } from "react-bootstrap";

import { useDispatch, useSelector } from "react-redux";

import { CTAButton } from "../CustomButtons";
import BeKindImage from "../../images/PrimaryHeroContentBgImg.png";
import "./index.scss";

function PrimaryHero() {
  const dispatch = useDispatch();
  const { t, i18n } = useTranslation();
  const isSignedIn = useSelector((state) => state.user.isSignedIn);

  return (
    <section className="snapping-section primary-hero-img-bg container-fluid">
      <Container className="primary-hero-content-container d-flex align-items-center">
        <Row className="primary-hero-content-row m-0">
          <Col
            xs={12}
            sm={12}
            md={6}
            className="first-column p-0 pb-2 pb-sm-2 pb-md-0 d-flex flex-column justify-content-center align-items-start"
          >
            <h1>Love Thy Neighbor</h1>
            <p>{t("home_slogan")}</p>
            {!isSignedIn && (
              <CTAButton onClick={() => dispatch({ type: "signUp" })}>
                {t("home_join_us_button")}
              </CTAButton>
            )}
          </Col>
          <Col xs={12} sm={12} md={6} className="px-0 ps-0 ps-sm-0 ps-md-3 ">
            <img src={BeKindImage} alt="img" className="img-fluid" />
          </Col>
        </Row>
      </Container>
    </section>
  );
}

export default PrimaryHero;
