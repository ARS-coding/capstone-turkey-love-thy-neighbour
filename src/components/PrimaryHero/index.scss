@import "../../style/variables";

.primary-hero-img-bg {
  background-image: url("../../images/PrimaryHeroBgImg.png");

  h1 {
    color: $logo-blue;
    font-size: max(40px);
  }

  p {
    color: white;
    font-size: 20px;
  }

  img {
    width: 100%;
    max-height: 350px;
    @media screen and (max-width: 767px) {
      max-height: 280px;
    }
  }

  .first-column {
    @media screen and (max-width: 768px) {
      padding-bottom: 50px !important;
    }
  }
}
