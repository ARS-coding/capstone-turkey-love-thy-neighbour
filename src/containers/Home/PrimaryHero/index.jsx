import React from "react";

import { Container, Row, Col } from "react-bootstrap";
import styled from "styled-components";
import CustomButton from "../../../components/CustomButton";

import "./index.scss";
import img from "../../../images/PrimaryHeroContentBgImg.png";

const CTAButton = styled(CustomButton)`
  color: var(--white);
  padding: 13px 65px;
  margin-top: 13px;
`;

function PrimaryHero() {
  return (
    <section className="snapping-section primary-hero-img-bg container-fluid">
      <Container className="primary-hero-content-container d-flex align-items-center">
        <Row className="primary-hero-content-row m-0">
          <Col
            xs={12}
            sm={12}
            md={6}
            className="first-column p-0 pb-xs-5 pb-md-5 d-flex flex-column justify-content-center align-items-start"
          >
            <h1>Love Thy Neighbor!</h1>
            <p>
              Letâ€™s build a strong, dynamic and caring community that uplifts
              each other.
            </p>
            <CTAButton className="cta-button">Join Us!</CTAButton>
            {/* TODO: Show sign up pop up when clicked */}
          </Col>
          <Col xs={12} sm={12} md={6} className="p-0 px-xs-1 px-md-1 h-100">
            <img src={img} alt="img" className="img-fluid" />
          </Col>
        </Row>
      </Container>
    </section>
  );
}

export default PrimaryHero;
