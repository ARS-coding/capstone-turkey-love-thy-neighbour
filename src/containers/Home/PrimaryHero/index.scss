@import "../../../style/variables";

.primary-hero-img-bg,
.primary-hero-content-container {
  height: calc(100vh - 60px);
}

.primary-hero-content-row,
.primary-hero-content-row > .col {
  width: 100%;
}

.primary-hero-img-bg {
  background-image: url("../../../utils/PrimaryHeroBgImg.png");

  h1 {
    color: $logo-blue;
    font-size: max(40px);
  }

  p {
    color: white;
    font-size: 20px;
  }

  img {
    width: 100%;
    height: 100%;
  }

  .first-column {
    @media screen and (max-width: 768px) {
      padding-bottom: 50px !important;
    }
  }

  .cta-button {
    background-color: $dark-blue;
  }
}
