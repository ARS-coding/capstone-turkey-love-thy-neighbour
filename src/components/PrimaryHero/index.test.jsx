import React from "react";
import renderer from "react-test-renderer";
import configureStore from "redux-mock-store";
import { Provider } from "react-redux";
import { BrowserRouter as Router } from "react-router-dom";
import PrimaryHero from ".";

describe("PrimaryHero page", () => {
  const mockState = {
    user: {
      isSignedIn: false,
    },
  };
  const mockStore = configureStore();
  let store;
  let tree;

  beforeEach(() => {
    store = mockStore(mockState);
    tree = renderer
      .create(
        <Provider store={store}>
          <PrimaryHero />
        </Provider>
      )
      .toJSON();
  });

  it("renders PrimaryHero page correctly", () => {
    expect(tree).toMatchSnapshot();
  });
});
