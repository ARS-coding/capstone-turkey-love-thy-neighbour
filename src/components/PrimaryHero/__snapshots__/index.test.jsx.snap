// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrimaryHero page renders PrimaryHero page correctly 1`] = `
<section
  className="snapping-section primary-hero-img-bg container-fluid"
>
  <div
    className="primary-hero-content-container d-flex align-items-center container"
  >
    <div
      className="primary-hero-content-row m-0 row"
    >
      <div
        className="first-column p-0 pb-2 pb-sm-2 pb-md-0 d-flex flex-column justify-content-center align-items-start col-md-6 col-sm-12 col-12"
      >
        <h1>
          Love Thy Neighbor!
        </h1>
        <p>
          Letâ€™s build a strong, dynamic and caring community that uplifts each other.
        </p>
        <button
          className="sc-bdnxRM sc-gtsrHT fhyKDU bLMOay"
          onClick={[Function]}
        >
          Join Us!
        </button>
      </div>
      <div
        className="px-0 ps-0 ps-sm-0 ps-md-3  col-md-6 col-sm-12 col-12"
      >
        <img
          alt="img"
          className="img-fluid"
          src="PrimaryHeroContentBgImg.png"
        />
      </div>
    </div>
  </div>
</section>
`;
